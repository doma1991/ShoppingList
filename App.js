import React, { useState, useEffect } from 'react';
import { View, Text, StyleSheet, FlatList, Alert } from 'react-native';
import Header from './components/Header';
import ListItem from './components/ListItem';
import AddItem from './components/AddItem';
import { v4 as uuidv4 } from 'uuid';
import AsyncStorage from '@react-native-community/async-storage';

const App = () => {

  const [items, setItems] = useState([
    {id: uuidv4(), text: "First Item"}
  ]);

  const saveData = async () => {
    try {
      await AsyncStorage.setItem("KEY", JSON.stringify(items))
    } catch (e) {
      Alert.alert('Failed to save the data to the storage.')
    }
  }

  const readData = async () => {
    try {
      const userData = await AsyncStorage.getItem("KEY")
      if (userData !== null) {
      setItems(JSON.parse(userData));
      }
    } catch (e) {
      Alert.alert('Failed to fetch the data from storage.')
    }
  }

  useEffect(() => {
    readData()
  }, [])

  const removeData = async () => {
    try {
      await AsyncStorage.removeItem("KEY");
    } catch (error) {
      Alert.alert('Failed to delete data from storage.')
    }
  }

  const deleteItem = (id) => {
    setItems(prevItems => {
      return prevItems.filter(item => item.id != id);
    });
    saveData();
  }

  const addItem = (text) => {
    if (!text) {
      Alert.alert('Error', 'Please enter an item.', [{ text: 'OK' }]);
    } else {
      setItems(prevItems => { return [{ id: uuidv4(), text }, ...prevItems]; })
      saveData();
    }
  }

  return (
    <View style={styles.container}>
      <Header title='Shopping List' />
      <AddItem addItem={addItem} />
      <FlatList data={items} renderItem={({ item }) => <ListItem item={item} deleteItem={deleteItem} />} />
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 60,
  },
})

export default App;