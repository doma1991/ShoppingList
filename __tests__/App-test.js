/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';
import Header from '../components/Header';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('Renders correctly', () => {
  renderer.create(<App />);
});

it('Renders Header component correctly', () => {
  const header = renderer.create(<Header />).toJSON();
  expect(header).toMatchSnapshot();
});
