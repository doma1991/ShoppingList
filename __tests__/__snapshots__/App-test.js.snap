// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays a header called Shopping List 1`] = `
<View
  style={
    Object {
      "backgroundColor": "darkslateblue",
      "height": 60,
      "padding": 15,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 23,
        "textAlign": "center",
      }
    }
  >
    Shopping List
  </Text>
</View>
`;
